src/components/GalleryList/ImageCard.tsx

```diff
@@ -0,0 +1,92 @@
+import React, { useState } from "react";
+import { Link } from "react-router-dom";
+
+import Button from "../Button";
+import { Photo } from "../../types";
+
+interface ImageCardProps {
+  photo: Photo;
+  isLike: boolean;
+  addToFavourites: (photo: Photo) => void;
+}
+
+const ImageCard = ({ photo, isLike, addToFavourites }: ImageCardProps) => {
+  const [hovered, setHovered] = useState<boolean>(false);
+
+  return (
+    <div
+      className="relative w-full h-full overflow-hidden"
+      onMouseEnter={() => setHovered(true)}
+      onMouseLeave={() => setHovered(false)}
+    >
+      <div className="w-full">
+        <Link to={`/photo/${photo.id}`} className="flex h-full">
+          <img
+            src={photo.url}
+            alt={photo.title}
+            className="w-full h-full object-cover"
+          />
+        </Link>
+      </div>
+      {hovered && (
+        <div className="w-full h-full">
+          <ul className="absolute z-2 top-3 right-4 flex gap-2 list-none">
+            <li>
+              <Button
+                className="bg-white flex items-center justify-center w-40 h-40 rounded-10"
+                label={
+                  <img
+                    src="/assets/copy.svg"
+                    alt="Copy icon"
+                    className="w-24 h-24 text-textGray"
+                  />
+                }
+                onClick={() => {}}
```

Comment:
We can make the onClick props as optional in button component instead of doing this.